pkg/storage/scan.go

```
@@ -40,52 +77,73 @@ func (s *PostgreSQLStore) ClearScansTable() error {
 }
 
 func (s *PostgreSQLStore) CreateScan(sc *domain.Scan) (*domain.Scan, error) {
+	tx, err := s.db.Begin()
+	if err != nil {
+		return nil, fmt.Errorf("failed to start transaction %w", err)
+	}
+	defer tx.Rollback()
 
-	status, _ := json.Marshal(sc.HostsStatus)
 	query := `
-    INSERT INTO scans (id, status, created_at, updated_at)
+    INSERT INTO scans ( tenant_id, operator_id, started_at, ended_at)
     values ($1, $2, $3, $4)
-    RETURNING id, created_at, updated_at`
+    RETURNING *`
 
-	rows, err := s.db.Query(query, sc.ID, status, sc.CreatedAt, sc.UpdatedAt)
-	if err != nil {
-		return nil, fmt.Errorf("error creating scan: `%v`", err)
+	row := tx.QueryRow(query, sc.TenantID, sc.OperatorID, sc.StartedAt, sc.EndedAt)
+	newScan := &domain.Scan{}
+	if err := scanIntoScan(row, newScan); err != nil {
+		return nil, fmt.Errorf("failed to insert scan: %w", err)
 	}
-
-	for rows.Next() {
-		return scanIntoScan(rows)
+	sc.ID = newScan.ID
+	errInsertScanHost := s.InsertScanHost(tx, sc)
+	if errInsertScanHost != nil {
+		return nil, errInsertScanHost
 	}
 
-	return nil, fmt.Errorf("error creating Scan")
+	if err := tx.Commit(); err != nil {
+		return nil, fmt.Errorf("failed to commit transaction: %w", err)
+	}
+	return newScan, nil
 }
 
-func scanIntoScan(rows *sql.Rows) (*domain.Scan, error) {
+func (s *PostgreSQLStore) InsertScanHost(tx *sql.Tx, sc *domain.Scan) error {
+	query := `
+    INSERT INTO scans_hosts (scan_id, host_id,status)
+    values ($1, $2, $3)`
 
-	scan := new(domain.Scan)
-	cols, _ := rows.Columns()
-	columns := make([]interface{}, len(cols))
-	columnPointers := make([]interface{}, len(cols))
-	for i := range columns {
-		columnPointers[i] = &columns[i]
+	for _, hostId := range sc.HostIDs {
```

If `sc.HostIDs` is empty, this method silently succeeds, which may not be desirable. Explicitly handle this case to avoid confusion.

---

Ok Lucas